server/py/services/api/api/endpoints/artifacts_v2.py

```diff
@@ -268,6 +268,19 @@ async def get_artifact(
             auth_info,
         )
     )
+
+    # Older clients (pre-1.8.0) do not support parsing "uid" and treat "tree^uid" as a single "tree" value.
+    # To ensure compatibility, we split "tree" here to extract "uid" if it exists.
+    if tree and "^" in tree:
+        tree, uri_object_uid = tree.split("^", 1)
+        if object_uid and object_uid != uri_object_uid:
+            mlrun.utils.logger.warning(
+                f"Conflicting UIDs detected: Provided object_uid='{object_uid}' "
+                f"does not match the UID extracted from the URI='{uri_object_uid}'."
+            )
```

Comment:
note this is server-side log line, thus suggested to format it in a structural way
```suggestion
            mlrun.utils.logger.warning(
                "Conflicting UIDs detected", object_uid=object_uid, extracted_object_uid=uri_object_uid
            )
```